lua/metals.lua

```diff
@@ -236,9 +237,19 @@ M.new_scala_project = function()
 end
 
 M.quick_worksheet = function()
-  local dir = "file://" .. fn.expand("%:p:h")
+  local dir = fn.expand("%:p:h")
   local name = fn.expand("%:p:h:t")
-  M.new_scala_file(dir, name, "scala-worksheet")
+  local path = Path:new(dir, name .. ".worksheet.sc")
+  if path:exists() then
+    local cmd = ":e" .. path.filename
+    local did_not_open, _ = pcall(vim.cmd(cmd))
+    if did_not_open then
+      log.error_and_show(string.format("Could not open file: %s", path.filename))
+    end
+  else
+    local dir_uri = "file://" .. dir
+    M.new_scala_file(dir_uri, name, "scala-worksheet")
```

Comment:
> in reality we could just create a worksheet manually without even calling Metals.

Turned out it's not that easy, since we still need to save this new file and then send `textDocument/didOpen` (and probably `metals/didFocusTextDocument`) notification for Metals to instantly pick up the worksheet. So, using Metals to create the file seems to be a okay 🤔

Reply:
Ah true, good call.